### DISCUSSION TITLE
`getMap` for AtomFamily

### DISCUSSION THREAD

**Discussion by sowhatdoido** [pmndrs/jotai#2070] [General]
Sorry if this has been asked before: I was looking for a way to iterate over an atom family.

I came upon a discussion where you said it isn't[ iterable by design](https://github.com/pmndrs/jotai/discussions/1912), but in another place you mentioned that AtomFamily is supposed to be a [simple helper function around map](https://github.com/pmndrs/jotai/discussions/1323#discussioncomment-3303035).

Is there a reason, philosophically, why you couldn't implement a `getMap` function that just returns the underlying map for people to iterate over?

https://github.com/pmndrs/jotai/blob/main/src/vanilla/utils/atomFamily.ts

```
createAtom.getMap = () => atoms;
```

I figured that would be a straight forward way of exposing the underlaying data structure for people to manipulate directly, including iterate over, but I wasn't sure if that would mess something up with reactivity or something.

**Comment by dai-shi** [comment]
See also #2056
